rockin/plugins/drilling_machine/drilling_machine_simulation_thread.cpp

```diff
@@ -0,0 +1,145 @@
+/***************************************************************************
+ *  drilling_machine_simulation_thread.cpp - Thread to simulate the drilling machine
+ *
+ *  Created: Mon Oct 5 12:25:11 2015
+ *  Copyright  2015 Alexander Moriarty
+ ****************************************************************************/
+
+/*  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Library General Public License for more details.
+ *
+ *  Read the full text in the LICENSE.GPL file in the doc directory.
+ */
+
+#include "drilling_machine_simulation_thread.h"
+
+#include <core/threading/mutex_locker.h>
+#include <boost/lexical_cast.hpp>
+#include <boost/thread/thread.hpp>
+
+/** @class DrillingMachineSimulationThread
+ * Thread to simulate RoCKIn@Work drilling machine.
+ * @author Alexander Moriarty
+ */
+fawkes::Mutex DrillingMachineSimulationThread::drill_machine_mutex_;
+
+/** Constructor. */
+DrillingMachineSimulationThread::DrillingMachineSimulationThread() :
+    Thread("DrillingMachineSimulationThread", Thread::OPMODE_CONTINUOUS),
+    cfg_timer_interval_(40)
+{
+}
+
+void DrillingMachineSimulationThread::init()
+{
+    try
+    {
+        cfg_timer_interval_ = config->get_uint("/llsfrb/clips/timer-interval");
+    } catch (fawkes::Exception &e)
+    {
+        // use default value
+    }
+
+    last_sent_command_timestamp_ = boost::posix_time::microsec_clock::local_time();
+
+    logger->log_info("DrillingMachineSim", "Plugin is in simulation mode.");
+
+    fawkes::MutexLocker lock2(&drill_machine_mutex_);
+
+    // First time
+    if (!last_status_msg_.has_state())
```

Comment:
Why is this conditional? It's probably a good idea to always initialize the state (in the init function).